CMakeLists.txt

```
@@ -0,0 +1,204 @@
+    "NOT CMAKE_SYSTEM_NAME MATCHES Darwin; NOT ANDROID" OFF)
+option(MOLD_USE_SYSTEM_MIMALLOC "Use system or vendored mimalloc" ON)
+option(MOLD_USE_SYSTEM_TBB "Use system or vendored TBB" ON)
+
+# Setup mimalloc
+if(MOLD_USE_MIMALLOC)
+    if(MOLD_USE_SYSTEM_MIMALLOC)
+        find_package(mimalloc REQUIRED)
+    else()
+        function(mold_add_mimalloc)
+            option(MI_BUILD_TESTS "Build test executables" OFF)
+            add_subdirectory(third-party/mimalloc EXCLUDE_FROM_ALL)
+            target_compile_definitions(mimalloc PRIVATE MI_USE_ENVIRON=0)
+        endfunction()
+
+        mold_add_mimalloc()
+    endif()
+
+    add_library(mimalloc::mimalloc ALIAS mimalloc)
+endif()
+
+# Setup TBB
+if(MOLD_USE_SYSTEM_TBB)
+    find_package(TBB REQUIRED)
+else()
+    function(mold_add_tbb)
+        set(CMAKE_BUILD_TYPE RelWithDebInfo)
+        set(BUILD_SHARED_LIBS OFF)
+        option(TBB_TEST "Enable testing" OFF)
+        string(APPEND CMAKE_CXX_FLAGS -D__TBB_DYNAMIC_LOAD_ENABLED=0)
+        option(TBB_STRICT "Treat compiler warnings as errors" OFF)
+        add_subdirectory(third-party/tbb EXCLUDE_FROM_ALL)
+    endfunction()
+
+    mold_add_tbb()
+endif()
+
+include(CheckPIESupported)
+check_pie_supported()
+
+add_library(mold-wrapper SHARED)
+set_target_properties(mold-wrapper PROPERTIES
+    POSITION_INDEPENDENT_CODE ON
+    PREFIX "") # Remove the default `lib` prefix
+target_link_libraries(mold-wrapper PRIVATE ${CMAKE_DL_LIBS})
+target_sources(mold-wrapper PRIVATE elf/mold-wrapper.c)
+
+add_executable(mold)
+target_compile_features(mold PRIVATE cxx_std_20)
+target_include_directories(mold SYSTEM PRIVATE third-party)
+
+if(NOT MSVC)
+    include(CheckLibraryExists)
+    check_library_exists(m pow "" LIBM_FOUND)
+
+    if(LIBM_FOUND)
+        target_link_libraries(mold PRIVATE m)
+    endif()
+
+    find_package(ZLIB REQUIRED)
+    target_link_libraries(mold PRIVATE ZLIB::ZLIB)
+endif()
+
+target_link_libraries(mold PRIVATE
+    TBB::tbb)
+
+if(MOLD_USE_MIMALLOC)
+    target_link_libraries(mold PRIVATE mimalloc::mimalloc)
+
+    if(MOLD_USE_SYSTEM_MIMALLOC)
+        target_compile_definitions(mimalloc INTERFACE USE_SYSTEM_MIMALLOC)
+    endif()
+endif()
+
+if(NOT CMAKE_SYSTEM_NAME MATCHES Darwin)
+    find_package(OpenSSL REQUIRED COMPONENTS Crypto)
+    target_link_libraries(mold PRIVATE OpenSSL::Crypto)
+endif()
+
+if(CMAKE_SYSTEM_PROCESSOR MATCHES riscv64
+    OR CMAKE_SYSTEM_PROCESSOR MATCHES armv6)
+    find_library(ATOMIC_LIBRARIES atomic REQUIRED)
+    target_link_libraries(mold PRIVATE ${ATOMIC_LIBRARIES})
+endif()
+
+file(STRINGS .git/HEAD GIT_HASH)
+
+if(GIT_HASH MATCHES "ref: ")
+    string(REPLACE "ref: " "" GIT_HASH "${GIT_HASH}")
+    file(STRINGS .git/${GIT_HASH} GIT_HASH)
+endif()
+
+set_property(SOURCE main.cc macho/output-chunks.cc APPEND PROPERTY
+    COMPILE_DEFINITIONS MOLD_VERSION="${CMAKE_PROJECT_VERSION}")
+set_property(SOURCE main.cc APPEND PROPERTY
+    COMPILE_DEFINITIONS GIT_HASH="${GIT_HASH}")
```

```suggestion
    COMPILE_DEFINITIONS "MOLD_VERSION=\"${CMAKE_PROJECT_VERSION}\"")
set_property(SOURCE main.cc APPEND PROPERTY
    COMPILE_DEFINITIONS "GIT_HASH=\"${GIT_HASH}\"")
```

Please do not use make-ish [legacy quoting](https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#:~:text=Note%20To%20support-,legacy%20CMake%20code,-%2C%20unquoted%20arguments%20may).

---

Done, thanks!